dashboard/views.py

```diff
@@ -330,12 +323,13 @@ def resource_access_within_week(request, course_id=0):
 
     output_df['resource_name'] = output_df.apply(
         lambda row:
-            RESOURCE_URLS[row.resource_type]["prefix"] +
+            (RESOURCE_ACCESS_CONFIG.get(row.resource_type).get("urls").get("prefix") +
             row.resource_id_part +
-            RESOURCE_URLS[row.resource_type]["postfix"] +
+            RESOURCE_ACCESS_CONFIG.get(row.resource_type).get("urls").get("postfix") +
             CANVAS_FILE_ID_NAME_SEPARATOR +
             row.resource_name_part + CANVAS_FILE_ID_NAME_SEPARATOR +
-            RESOURCE_VALUES[INVERSE[row.resource_type]]['icon'],
+            RESOURCE_VALUES.get(RESOURCE_VALUES_MAP.get(row.resource_type)).get('icon')
+            ),
         axis=1)
     output_df.drop(columns=['resource_id_part', 'resource_name_part', 'resource_id_name'], inplace=True)
```

Comment:
Yeah, this block isn't very read-able, in general, but we can work on that in the future.